src/facade/dragonfly_connection.cc

```diff
@@ -39,6 +39,9 @@ ABSL_FLAG(std::uint64_t, request_cache_limit, 1ULL << 26,  // 64MB
 
 ABSL_FLAG(bool, no_tls_on_admin_port, false, "Allow non-tls connections on admin port");
 
+ABSL_FLAG(std::uint64_t, pipeline_squash, 0,
+          "Number of queued pipelined commands above which squashing is enabled, 0 means disabled");
+
```

Comment:
I've added this flag, so its disabled for now. Not sure if we should use it 🤷🏻 

Two issues:
1. It reorders execution and as there is no transaction that locks all keys (like in multi), this re-ordering can be observed
2. I'm not totally sure we don't find any issues connected with it

Reply:
on second thoughts, https://redis.io/docs/manual/pipelining/ it implies that there is an ordering. There must be something I am missing here

Reply:
Yes, its regular squashing 🙂 It means that we split the commands by shards and execute them in parallel. Relative to every key, the order of commands is preserved, so the outcome is always the same. The _only_ thing that can happen is that

```
SET A 1
SET B 1
SET C 1
```

will execute in random order if A, B, C belong to different threads because it parallelized.

Reply:
No, that's what the capturing reply builder is about - it records replies and then they're reordered (by the multi command squasher)